addons/website/static/src/js/menu/content.js

```diff
@@ -1072,8 +1072,90 @@ function _clonePage(pageId) {
     });
 }
 
+const KeyboardNavigationMixin = require('web.KeyboardNavigationMixin');
+const KeyboardNavigationWidget = Widget.extend(KeyboardNavigationMixin, {
+    /**
+     * @override
+     */
+    init() {
+        this.isFullscreen = false;
+        return KeyboardNavigationMixin.init.call(this);
+    },
+    /**
+     * @override
+     */
+    start() {
+        const ret = this._super(...arguments);
+        this.setElement($('body'));
+        return ret;
+    },
+
+    //--------------------------------------------------------------------------
+    // Private
+    //--------------------------------------------------------------------------
+
+    /**
+     * Toggles the fullscreen mode.
+     *
+     * @private
+     * @param {Boolean} state toggle fullscreen on/off (true/false)
+     */
+    async _toggleFullscreen(state) {
+        this.isFullscreen = state;
+        const dummy = document.createElement('div');
+        const topMenu = document.querySelector('#web_editor-top-edit') || dummy;
+        const leftPanel = document.querySelector('#oe_snippets') || dummy;
+        const navbar = document.querySelector('#oe_main_menu_navbar') || dummy;
+        [topMenu, leftPanel, navbar, document.body].forEach(el => el.style.transition = 'all 0.25s');
+        if (this.isFullscreen) {
+            document.body.style.setProperty('padding', '0', 'important');
+            topMenu.style.setProperty('top', `-${window.getComputedStyle(topMenu).height}`, 'important');
+            navbar.style.setProperty('top', `-${window.getComputedStyle(navbar).height}`, 'important');
+            leftPanel.style.setProperty('left', `-${window.getComputedStyle(leftPanel).width}`, 'important');
+        } else {
+            document.body.style.removeProperty('padding');
+            topMenu.style.removeProperty('top');
+            navbar.style.removeProperty('top');
+            leftPanel.style.removeProperty('left');
+        }
+        let resizing = true;
+        const resizeFunction = () => {
+            if (resizing) {
+                window.dispatchEvent(new Event('resize'));
+                window.requestAnimationFrame(resizeFunction);
+            }
+        };
+        window.requestAnimationFrame(resizeFunction);
+        document.body.addEventListener('transitionend', async ev => {
+            if (ev.target.matches('body') && ev.propertyName === 'padding-top') {
+                resizing = false;
+                [topMenu, leftPanel, navbar, document.body].forEach(el => el.style.removeProperty('transition'));
+            }
+        });
+        return;
+    },
+
+    //--------------------------------------------------------------------------
+    // Handlers
+    //--------------------------------------------------------------------------
+
+    /**
+     * Toggles "fullscreen" mode.
+     *
+     * @override
+     */
+    _onKeyDown(ev) {
+        // keyCode 27 => Esc.
+        if (!(ev.keyCode === 27) || ev.target.matches('.modal')) {
```

Comment:
- `$.ui.keyCode.ESCAPE`, something like that
- Heuuu, `a !== b` instead of `!(a === b)` ? ^^